plugins/modules/df_deployment.py

```diff
@@ -0,0 +1,467 @@
+  returned: when supported
+  type: str
+sdk_out_lines:
+  description: Returns a list of each line of the captured CDP SDK log.
+  returned: when supported
+  type: list
+  elements: str
+'''
+
+
+class DFDeployment(CdpModule):
+    def __init__(self, module):
+        super(DFDeployment, self).__init__(module)
+
+        # Set variables
+        self.name = self._get_param('name')
+        self.dep_crn = self._get_param('dep_crn')
+        self.df_crn = self._get_param('df_crn')
+        self.df_name = self._get_param('df_name')
+        self.flow_ver_crn = self._get_param('flow_ver_crn')
+        self.flow_name = self._get_param('flow_name')
+        self.flow_ver = self._get_param('flow_ver')
+        self.size = self._get_param('size')
+        self.static_node_count = self._get_param('static_node_count')
+        self.autoscale_enabled = self._get_param('autoscale_enabled')
+        self.autoscale_nodes_min = self._get_param('autoscale_nodes_min')
+        self.autoscale_nodes_max = self._get_param('autoscale_nodes_max')
+        self.nifi_ver = self._get_param('nifi_ver')
+        self.autostart_flow = self._get_param('autostart_flow')
+        self.parameter_groups = self._get_param('parameter_groups')
+        self.kpis = self._get_param('kpis')
+
+        self.state = self._get_param('state')
+        self.wait = self._get_param('wait')
+        self.delay = self._get_param('delay')
+        self.timeout = self._get_param('timeout')
+
+        # Initialize return values
+        self.deployment = None
+        self.changed = False
+
+        # Initialize internal values
+        self.target = {}
+
+        # Execute logic process
+        self.process()
+
+    @CdpModule._Decorators.process_debug
+    def process(self):
+        # Prepare information
+        if self.df_crn is None:
+            self.df_crn = self.cdpy.df.resolve_service_crn_from_name(self.df_name)
+            if self.df_crn is None:
+                self.module.fail_json(
+                    msg="Either df_crn must be supplied or resolvable from df_name")
+        if self.dep_crn is not None:
+            self.target = self.cdpy.df.describe_deployment(dep_crn=self.dep_crn)
+        elif self.name is not None and self.df_crn is not None:
+            self.target = self.cdpy.df.describe_deployment(df_crn=self.df_crn, name=self.name)
+            if self.target is not None:
+                self.dep_crn = self.target['crn']
+        # Process execution
+        if self.target is not None:
+            # DF Deployment exists
+            if self.state in ['absent']:
+                # Existing Deployment to be removed
+                if self.module.check_mode:
+                    self.module.log(
+                        "Check mode enabled, skipping termination of Deployment %s" % self.dep_crn)
+                    self.deployment = self.target
+                else:
+                    self._terminate_deployment()
+            # Existing deployment to be retained
```

Comment:
This comment belongs below the `elif` (being consistent with the comment positioning above)